engine-tests/src/tests/xcc.rs

```diff
@@ -257,28 +256,6 @@ fn test_xcc_exec_gas() {
             outcome.burnt_gas,
             router_exec_cost
         );
-
-        assert_eq!(
```

Comment:
This section has been removed since the `VMOutcome` doesn't contain `action_receipts` anymore.

Reply:
Yes. There is a [PR](https://github.com/near/nearcore/pull/9332) about that. Try to find out.

Reply:
I guess you can collect receipts by using this [method](https://github.com/near/nearcore/blob/master/runtime/near-vm-runner/src/logic/dependencies.rs#L321) of this [trait](https://github.com/near/nearcore/blob/master/runtime/near-vm-runner/src/logic/dependencies.rs#L121). We already have a [type](https://github.com/aurora-is-near/aurora-engine/blob/develop/engine-tests/src/utils/mocked_external.rs#L24) which implements this trait.

Reply:
Looks like the `action_log` should be the right place to find this. I do not immediately see why the promises are not recorded. I think we should continue investigating this though.